libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2950,6 +2942,11 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 				"\"chainid\" is not supported by the VM version."
 			);
 	}
+	else if (m_currentContract || m_currentFreeFunction)
+		if (auto const* typeType = dynamic_cast<TypeType const*>(exprType))
+			if (auto const* contractType = dynamic_cast<ContractType const*>(typeType->actualType()))
+				if (&contractType->contractDefinition() != m_currentContract)
+					currentDependencyAnnotation()->contractDependencies.emplace(&contractType->contractDefinition(), &_memberAccess);
```

Comment:
So, according to https://github.com/ethereum/solidity/pull/10228#discussion_r591582547, I think this should be moved into the graph.

Or maybe just removed? I think that the special case for `creationCode` and `runtimeCode` you already added to the graph covers all member expressions that can result in a bytecode dependency.